scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -473,7 +473,7 @@ def device_is_available(self, instance):
                 d.counter_increment()
                 avail = True
                 logger.debug(f"Retain DUT:{d.platform}, Id:{d.id}, "
-                             f"counter:{d.counter}")
+                             f"counter:{d.counter}, failures:{d.failures}")
```

Comment:
Both are necessary, as hardware failure oft shows as test timeouts, which are `failed` statuses, not `error`s.
At the same time, I'd rank them separately; a separate `failures` and `errors` counters, which would be used in sorting, with errors given a higher priority (`sorted()`'s `key` accepts tuples)

Reply:
The proposed change counts both 'error' and 'failed' test instance statuses for each DUT as a failure situation happened there not going deeper into details of what was the origin. It is a single counter to avoid overcomplication, but to stimulate retry attempts on different DUTs choosing less 'suspicious' one as a candidate for the next round. This way it works for any root cause not known at the moment - either at a board, a harness, or a test, or any their combination.